eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -256,6 +261,10 @@ protected function indexIncrementally(
             $count = $this->gateway->countContentInSubtree($location->pathString);
             $generator = $this->gateway->getContentInSubtree($location->pathString, $iterationCount);
             $purge = false;
+        } elseif ($contentType = $input->getOption('content-type')) {
+            $count = $this->gateway->countContentWithContentTypeIdentifier($contentType);
+            $generator = $this->gateway->getContentWithContentTypeIdentifier($contentType, $iterationCount);
+            $purge = false;
```

Comment:
Given you can't add anything to that Gateway without breaking BC promise (see previous comment), I think better solution would be to either use Repository Filtering (`\eZ\Publish\API\Repository\ContentService::find`) or make a dedicated method in `\eZ\Publish\SPI\Persistence\Content\Handler`. The latter option would be more performant and require implementing cache layer (which is good).